toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1452,6 +1458,102 @@ func updateUserPassword(installRoot, username, password string) (err error) {
+
+	err = selinuxUpdateConfig(systemConfig, installChroot)
+	if err != nil {
+		logger.Log.Errorf("Failed to update SELinux config")
+		return
+	}
+	err = selinuxRelabelFiles(systemConfig, installChroot, mountPointToFsTypeMap)
+	if err != nil {
+		logger.Log.Errorf("Failed to label SELinux files")
+		return
+	}
+	return
+}
+
+func selinuxUpdateConfig(systemConfig configuration.SystemConfig, installChroot *safechroot.Chroot) (err error) {
+	const (
+		configFile     = "etc/selinux/config"
+		selinuxPattern = "^SELINUX=.*"
+	)
+	var mode string
+
+	switch systemConfig.KernelCommandLine.SELinux {
+	case configuration.SELinuxEnforcing, configuration.SELinuxForceEnforcing:
+		mode = configuration.SELinuxEnforcing.String()
+	case configuration.SELinuxPermissive:
+		mode = configuration.SELinuxPermissive.String()
+	}
+
+	selinuxConfigPath := filepath.Join(installChroot.RootDir(), configFile)
+	selinuxMode := fmt.Sprintf("SELINUX=%s", mode)
+	err = sed(selinuxPattern, selinuxMode, "`", selinuxConfigPath)
+	return
+}
+
+func selinuxRelabelFiles(systemConfig configuration.SystemConfig, installChroot *safechroot.Chroot, mountPointToFsTypeMap map[string]string) (err error) {
+	const (
+		squashErrors        = false
+		configFile          = "etc/selinux/config"
+		fileContextBasePath = "etc/selinux/%s/contexts/files/file_contexts"
+	)
+	var listOfMountsToLabel []string
+
+	// Search through all our mount points for supported filesystem types
+	// Note for the future: SELinux can support any of {btrfs, encfs, ext2-4, f2fs, jffs2, jfs, ubifs, xfs, zfs}, but the build system currently
+	//     only supports the below cases:
+	for mount, fsType := range mountPointToFsTypeMap {
+		switch fsType {
+		case "ext2", "ext3", "ext4":
+			listOfMountsToLabel = append(listOfMountsToLabel, mount)
+		case "fat32", "fat16", "vfat":
+			logger.Log.Debugf("SELinux will not label mount at (%s) of type (%s), skipping", mount, fsType)
+		default:
+			err = fmt.Errorf("Unknown fsType (%s) for mount (%s), cannot configure SELinux", fsType, mount)
+			return
+		}
+	}
+
+	// Find the type of policy we want to label with
+	selinuxConfigPath := filepath.Join(installChroot.RootDir(), configFile)
+	stdout, stderr, err := shell.Execute("sed", "-n", "s/^SELINUXTYPE=\\(.*\\)$/\\1/p", selinuxConfigPath)
+	if err != nil {
+		logger.Log.Errorf("Could not find an SELINUXTYPE in %s", selinuxConfigPath)
+		logger.Log.Error(stderr)
+		return
+	}
+	selinuxType := strings.TrimSpace(stdout)
+	fileContextPath := fmt.Sprintf(fileContextBasePath, selinuxType)
+
+	logger.Log.Debugf("Running setfiles to apply SELinux labels on mount points: %v", listOfMountsToLabel)
+	err = installChroot.UnsafeRun(func() error {
+		args := []string{"-m", "-v", fileContextPath}
+		args = append(args, listOfMountsToLabel...)
+
+		// We only want to print basic info, filter out the real output unless at trace level (Execute call handles that)
+		files := 0
+		lastFile := ""
```

Comment:
I don't think the value of lastFile is ever set before printing failure message?

Reply:
Hmm, looks like it. Let me figure out what I was intending here. I think I just forgot to pull the line from the onStdout() callback

Reply:
Resolved with the patch I sent to Chris.